website/docs/quickstart.md

```diff
@@ -0,0 +1,211 @@
+2. **Package Test**: Depends on **Package Build** - This is where the package is installed in a clean environment and tested to ensure it was built correctly -- for example, to ensure that package artifacts are installed in the proper locations and have the correct permissions.
+3. **Create Output Image** (optional) - This stage is independent of the first two; you may build a container image with Dalec *without* first building a package or running any tests; see [Container Only Builds](container-only-builds.md). However, for many use cases, a package will have been built in stage (1) and it will be installed in a base image for the resulting **output container image** to be created. There may be additional runtime dependencies specified in the spec file that are installed at this stage, and additional configuration of the image itself is also allowed. 
+
+## Creating a Package and Container from Source
+
+Now, without further ado, let's get started.
+
+To do our build, we need a few things:
+
+1. A list of sources to pull from
+2. A build script to build the sources
+3. A list of artifacts to include in the package
+
+In this example, we'll build the `go-md2man` package and container from the [`go-md2man`](https://github.com/cpuguy83/go-md2man) repo using `v2.0.3` tag in the repo.
+
+First, let's start with the constructing a [Dalec spec](spec.md) file.
+
+We define the metadata for the package in the spec. This includes the name, packager, vendor, license, website, and description of the package. You may notice that many of these fields appear in package manager metadata for rpm and deb packages. This is because, for certain targets, Dalec will generate package files for these package managers and will utilize this information for that purpose.
+
+```yaml
+# syntax=ghcr.io/azure/dalec/frontend:latest
+name: go-md2man
+version: 2.0.3
+revision: "1"
+license: MIT
+description: A tool to convert markdown into man pages (roff).
+packager: Dalec Example
+vendor: Dalec Example
+website: https://github.com/cpuguy83/go-md2man
+```
+
+:::tip
+In metadata section, `packager`, `vendor` and `website` are optional fields.
+:::
+
+In the next section of the spec, we define the [sources](sources.md) that we will be pulling from. In this case, we are pulling from a git repository.
+
+One thing to note: in many build systems you will not have access to the Internet while building the package, and indeed that is the case with the `mariner2` target. As such, this build will fail because `go build` will try to download the go modules. For this reason, we added a `generate` section to the source to run `go mod download` in a docker image with the `src` source mounted and then extract the go modules from the resulting filesystem.
```

Comment:
Add note that this is to ensure that the generated source package can be built without internet access and would be the default for all dalec targets.